larpandoracontent/LArTrackShowerId/TrackShowerIdFeatureTool.cc

```diff
@@ -298,6 +298,140 @@ StatusCode PfoHierarchyFeatureTool::ReadSettings(const TiXmlHandle /*xmlHandle*/
 //------------------------------------------------------------------------------------------------------------------------------------------
 //------------------------------------------------------------------------------------------------------------------------------------------
 
+ConeShapeFeatureTool::ConeShapeFeatureTool(): m_conMinHits(3), m_minCharge (0.1f), m_conFracRange(0.2f), m_MoliereRadius(10.1f), m_MoliereRadiusFrac(0.2f)
+{
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+void ConeShapeFeatureTool::Run(
+    LArMvaHelper::MvaFeatureVector &featureVector, const Algorithm *const pAlgorithm, const pandora::ParticleFlowObject *const pInputPfo)
+{
+    if (PandoraContentApi::GetSettings(*pAlgorithm)->ShouldDisplayAlgorithmInfo())
+        std::cout << "----> Running Algorithm Tool: " << this->GetInstanceName() << ", " << this->GetType() << std::endl;
+    
+    ClusterList clusterListW;
+    LArPfoHelper::GetClusters(pInputPfo, TPC_VIEW_W, clusterListW);
+
+    LArMvaHelper::MvaFeature haloTotalRatio, concentration, conicalness;
+    
+    if(!clusterListW.empty())
+    {    
+        CaloHitList clusterCaloHitList;
+        clusterListW.front()->GetOrderedCaloHitList().FillCaloHitList(clusterCaloHitList);
+       
+        CartesianVector pfoStart(0.f, 0.f, 0.f);
+    
+        try
+        {
+            pfoStart = LArPfoHelper::GetVertex(pInputPfo)->GetPosition();
+        }
+        catch (const StatusCodeException &)
+        {
+            pfoStart = clusterCaloHitList.front()->GetPositionVector();
+        }
+
+        CartesianVector centroid(0.f, 0.f, 0.f);
+        LArPcaHelper::EigenVectors eigenVecs;
+        LArPcaHelper::EigenValues eigenValues(0.f, 0.f, 0.f);
```

Comment:
Should the initialisations be something extreme (like numeric_limits::low)?  I feel like 0 is a potentially interesting value.